### PR TITLE
Feature 58 As a user I want to have an ability to record my screen and store recording locally

### PR DESCRIPTION
**Pull request by IvanitskyiVolodymyr**

### CODE REVIEW

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using System.Threading.Tasks;
+
+namespace Recodo.Desktop.Logic
+{
+    public class RecorderConfiguration
+    {
+        public bool IsAudioEnable { get; set; } = true;
+        public string SelectedAudioInputDevice { get; set; }
+        public string SelectedAudioOutputDevice { get; set; }
+        public string RecorderWindowTitle { get; set; }
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/RecorderConfiguration.cs:16]: Can it handle screen selection?

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderHelper.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Recodo.Desktop.Logic
+{
+    public class RecorderHelper
+    {
+        static RecorderHelper()
+        {
+            resolutions = new Dictionary<string, Tuple<int, int>>();
+            resolutions.Add("2160p 4K", Tuple.Create(3840, 2160));
+            resolutions.Add("1440p HD", Tuple.Create(2560, 1440));
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/RecorderHelper.cs:13]: Do you check if user has such resolution?)

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        {
+            var sources = new List<RecordingSourceBase>();
+            sources.AddRange(Recorder.GetDisplays());
+
+            return sources;
+        }
+
+        public List<string> GetWindows()
+        { 
+            return Recorder.GetWindows().Select(w => w.Title).ToList();
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs:96]: Can't they have the same titles? If I open two windows of chrome

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+        public List<string> GetOutputAudioDevices()
+        {
+            return Recorder.GetSystemAudioDevices(AudioDeviceSource.OutputDevices)
+                ?.Select(d => d.FriendlyName)
+                ?.ToList();
+        }
+
+        public List<RecordingSourceBase> GetDisplays()
+        {
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs]: It seems that it's not used

File: RecodoDesktop/RecodoDesktop/Countdown.xaml.cs [hunk 5]
```diff
@@ -6,32 +7,35 @@ namespace Recodo.Desktop.Main
...
+        TimeSpan time = TimeSpan.FromSeconds(3);
+
+		public Countdown(RecorderService recorderService)
 		{
 			InitializeComponent();
-            TimerStart(finishTimerMethod);
+            _recorderService = recorderService;
+            Timer = new DispatcherTimer();
+            Timer.Interval = new System.TimeSpan(0, 0, 1);
+            Timer.Tick += Timer_Tick;
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/RecodoDesktop/Countdown.xaml.cs]: Check tabulation

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderConfiguration.cs (hunk 1 above)
Comment (IvanitskyiVolodymyr) [reply RecodoDesktop/Recodo.Desktop.Logic/RecorderConfiguration.cs:16]: so can

File: RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs (hunk 3 above)
Comment (IvanitskyiVolodymyr) [reply RecodoDesktop/Recodo.Desktop.Logic/RecorderService.cs:96]: So there will be the same two titles but it will be different objects

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by NickRost — APPROVED** [review]

**Pull request merged**